scripts/ta_bin_to_c.py

```diff
@@ -11,75 +11,81 @@
+                uncompressed_size = len(bytes)
+                if args.compress:
+                        bytes = zlib.compress(bytes)
+                size = len(bytes)
 
-	f = open(args.out, 'w')
-	f.write('/* Generated from ' + args.ta + ' by ' +
-		os.path.basename(__file__) + ' */\n\n')
-	f.write('#include <compiler.h>\n');
-        f.write('#include <kernel/early_ta.h>\n\n');
-	f.write('__extension__ const struct early_ta __early_ta_' +
-		ta_uuid.hex +
-		'\n__early_ta __aligned(__alignof__(struct early_ta)) = {\n')
-	f.write('\t.uuid = {\n')
-	f.write('\t\t.timeLow = 0x{:08x},\n'.format(ta_uuid.time_low))
-	f.write('\t\t.timeMid = 0x{:04x},\n'.format(ta_uuid.time_mid))
-	f.write('\t\t.timeHiAndVersion = ' +
-		'0x{:04x},\n'.format(ta_uuid.time_hi_version))
-	f.write('\t\t.clockSeqAndNode = {\n')
-	csn = '{0:02x}{1:02x}{2:012x}'.format(ta_uuid.clock_seq_hi_variant,
-		ta_uuid.clock_seq_low, ta_uuid.node)
-	f.write('\t\t\t')
-	f.write(', '.join('0x' + csn[i:i+2] for i in range(0, len(csn), 2)))
-	f.write('\n\t\t},\n\t},\n')
-	f.write('\t.size = {:d},\n'.format(size))
-	if args.compress:
-		f.write('\t.uncompressed_size = '
-			'{:d},\n'.format(uncompressed_size))
-	f.write('\t.ta = {\n')
-	i = 0
-	while i < size:
-		if i % 8 == 0:
-			f.write('\t\t');
-		f.write('0x' + '{:02x}'.format(ord(bytes[i])) + ',')
-		i = i + 1
-		if i % 8 == 0 or i == size:
-			f.write('\n')
-		else:
-			f.write(' ')
-	f.write('\t},\n')
-	f.write('};\n');
-	f.close()
+        f = open(args.out, 'w')
+        f.write('/* Generated from ' + args.ta + ' by ' +
+                os.path.basename(__file__) + ' */\n\n')
+        f.write('#include <compiler.h>\n')
+        f.write('#include <kernel/early_ta.h>\n\n')
+        f.write('__extension__ const struct early_ta __early_ta_' +
+                ta_uuid.hex +
+                '\n__early_ta __aligned(__alignof__(struct early_ta)) = {\n')
+        f.write('\t.uuid = {\n')
+        f.write('\t\t.timeLow = 0x{:08x},\n'.format(ta_uuid.time_low))
+        f.write('\t\t.timeMid = 0x{:04x},\n'.format(ta_uuid.time_mid))
+        f.write('\t\t.timeHiAndVersion = ' +
+                '0x{:04x},\n'.format(ta_uuid.time_hi_version))
+        f.write('\t\t.clockSeqAndNode = {\n')
+        csn = '{0:02x}{1:02x}{2:012x}'.format(ta_uuid.clock_seq_hi_variant,
+                                              ta_uuid.clock_seq_low,
+                                              ta_uuid.node)
+        f.write('\t\t\t')
+        f.write(', '.join('0x' + csn[i:i+2] for i in range(0, len(csn), 2)))
+        f.write('\n\t\t},\n\t},\n')
+        f.write('\t.size = {:d},\n'.format(size))
+        if args.compress:
+                f.write('\t.uncompressed_size = '
+                        '{:d},\n'.format(uncompressed_size))
+        f.write('\t.ta = {\n')
+        i = 0
+        while i < size:
+                if i % 8 == 0:
+                        f.write('\t\t')
+                f.write('0x' + '{:02x}'.format(bytes[i]) + ',')
```

Comment:
Speaking about the format (regardless of Python 2 or Python 3), the way to put together the string here with mixing + and format is a bit odd. The old way of doing things in Python was with `+` or `... % (foo, bar)`. The new way is to just format. But then you construct the entire string in one go (and avoid use `+`). So this line for example should be like this:
```python
f.write("0x{:02x},".format(bytes[i]))
```
I don't say we need to change this now in this patch, since it's a different thing. But in the future if someone feel for doing an update, then please go a head.